pysat/instruments/methods/testing.py

```diff
@@ -29,3 +76,49 @@ def generate_fake_data(t0, num_array, period=5820, data_range=[0.0, 24.0],
         data = ((t0 + num_array) / period).astype(int)
 
     return data
+
+
+def generate_times(fnames, sat_id, freq='1S'):
+    """Construct list of times for simulated instruments
+
+    Parameters
+    ----------
+    fnames : (list)
+        List of filenames
+    sat_id : (str or NoneType)
+        Instrument satellite ID (accepts '' or a number (i.e., '10'), which
+        specifies the number of data points to include in the test instrument)
+    freq : string
+        Frequency of temporal output, compatible with pandas.date_range
+        [default : '1S']
+
+    Outputs
+    -------
+    uts : (array)
+        Array of integers representing uts for a given day
+    index : (DatetimeIndex)
+        The DatetimeIndex to be used in the pysta test instrument objects
+    date : (datetime)
+        The requested date reconstructed from the fake file name
+    """
+
+    # grab date from filename
+    parts = os.path.split(fnames[0])[-1].split('-')
```

Comment:
Could loop over fnames here then combine the indices together.

Reply:
To do this properly, we need an end-point in the fake file name.

Reply:
Only the filenames to be loaded for the given day are passed in. This is more likely a pysat 3.0 issue I just happened to see it while checking things out.

Reply:
Adding a note here for changes in #266.